src/openms/source/ANALYSIS/ID/FIAMSScheduler.cpp

```diff
@@ -0,0 +1,122 @@
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+// 
+// --------------------------------------------------------------------------
+// $Maintainer: Timo Sachsenberg $
+// $Authors: Svetlana Kutuzova, Douglas McCloskey $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/ID/FIAMSDataProcessor.h>
+#include <OpenMS/ANALYSIS/ID/FIAMSScheduler.h>
+
+#include <boost/algorithm/string.hpp>
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+namespace OpenMS {
+  /// default constructor
+  FIAMSScheduler::FIAMSScheduler(
+    String filename,
+    String base_dir,
+    bool load_cached
+  )
+    : 
+    filename_(filename),
+    base_dir_(base_dir),
+    load_cached_(load_cached),
+    samples_()
+  {
+  loadSamples_();
+  }
+
+  /// assignment operator
+  FIAMSScheduler& FIAMSScheduler::operator=(const FIAMSScheduler& rhs) {
+    if (this == &rhs) return *this;
+    filename_ = rhs.filename_;
+    base_dir_ = rhs.base_dir_;
+    load_cached_ = rhs.load_cached_;
+    samples_ = rhs.samples_;
+    return *this;
+  }
+
+  void FIAMSScheduler::loadSamples_() {
+    CsvFile csv_file(filename_, ',');
+    StringList headers;
+    csv_file.getRow(0, headers);
+    StringList row;
+    for (Size i = 1; i < csv_file.rowCount(); ++i) {
+      csv_file.getRow(i, row);
+      std::map<String, String> mapping;
+      for (Size j = 0; j < headers.size(); ++j) {
+        mapping[headers[j]] = row[j];
+      }
+      samples_.push_back(mapping);
+    }
+  }
+
+  void FIAMSScheduler::run() {
+    #pragma omp parallel for
+    for (size_t i = 0; i < samples_.size(); ++i) {
```

Comment:
this will not compile on Visual Studio 2017 and earlier. Please use a signed type 
(@timosachsenberg  @jpfeuffer  our CI should catch this... is the jenkins machine actually running on VS2015 as the name implies?)